client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -1008,6 +1014,15 @@ export class PlanFeatures2023Grid extends Component<
 	}
 }
 
+const withIsLargeCurrency = ( Component: LocalizedComponent< typeof PlanFeatures2023Grid > ) => {
+	return function ( props: PlanFeatures2023GridType ) {
+		const isLargeCurrency = useIsLargeCurrency(
+			( props.planProperties || [] ).map( ( properties ) => properties.planName as PlanSlug )
+		);
+		return <Component { ...props } isLargeCurrency={ isLargeCurrency } />;
+	};
+};
```

Comment:
was there no way to avoid the HOC? It should be doable if we refactor renderPlanPrice into a PlanPrice component, right?

p.s. I foresee we'll be doing a lot less prop drilling and more component extractions and hook usage on additional refactors.

Reply:
> was there no way to avoid the HOC? It should be doable if we refactor renderPlanPrice into a PlanPrice component, right?

That's what I had done earlier :) I had to revert that change and add the HOC via https://github.com/Automattic/wp-calypso/pull/75210/commits/8b539b9acc8401a39556966e4d8b1591642d3882

While investigating [this](https://github.com/Automattic/wp-calypso/pull/75210#pullrequestreview-1376074288) bug, I noticed that tablet view renders two separate tables:
https://github.com/Automattic/wp-calypso/blob/478ceeb1cb32c600c915733238765945f670f20f/client/my-sites/plan-features-2023-grid/index.tsx#L403-L412

So effectively, for tablet, the value would have to be calculated in the `renderTabletView` function and then passed down to `renderTable`. Similarly, for desktop, the value would have to be calculated in the `render` function itself.